Providers/Profile.swift

```diff
@@ -588,11 +606,11 @@ public class BrowserProfile: Profile {
         private let syncLock = NSRecursiveLock()
 
         var isSyncing: Bool {
-            syncLock.lock()
```

Comment:
Because of the multi-threaded nature of sync I think it's important to keep this lock here to make sure that we are reporting the correct `isSyncing` boolean value when sync is happening on the various background threads.